os/drivers/input/ist415.c

```diff
@@ -89,185 +98,1910 @@ struct touchscreen_ops_s g_ist415_ops = {
  ****************************************************************************/
 
 /****************************************************************************
- * Name: ist415_get_touch_data
+ * Name: put_unaligned_be32
  ****************************************************************************/
 
-static int ist415_get_touch_data(struct ist415_dev_s *dev, FAR void *buf)
+static __inline void put_unaligned_be32(uint32_t value, void *addr) {
+    uint8_t *p = (uint8_t *)addr;
```

Comment:
Instead of typecasting 32bit int to 8bit int and using array indices to change each bit.
I think it's better to do this in single move.